pkg/codegen/pcl/binder_resource.go

```diff
@@ -624,6 +624,12 @@ func (b *binder) bindResourceBody(node *Resource) hcl.Diagnostics {
 	// Bind the resource's body.
 	scopes := newResourceScopes(b.root, node, rangeKey, rangeValue)
 	block, blockDiags := model.BindBlock(node.syntax, scopes, b.tokens, b.options.modelOptions()...)
+	for _, diag := range blockDiags {
+		if b.options.skipResourceTypecheck && diag.Severity == hcl.DiagError {
+			// If we are skipping resource type-checking, convert errors to warnings.
```

Comment:
Are we ok doing this for all diagnostics? I'd have thought we'd plumb this down to the bindExpression parts that are just for type checking. There's other errors like cannotEvaluateAnonymousFunctionExpressions which maybe should still be errors.